### PR TITLE
dashing 0.3.0 (new formula)

### PR DESCRIPTION
**Pull request by ylluminarious**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
I changed the formula to not use `go_resource` blocks as per the discussion [here.](https://github.com/Homebrew/homebrew-core/pull/23696)

### CODE REVIEW

**Comment by ylluminarious** [comment]
@ilovezfs The test error appears to be completely unrelated to my formula. Pasting from the [Jenkins log:](https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/18480/version=high_sierra/console)

```
16:35:57 Started by upstream project "Homebrew Core Pull Requests" build number 18480
16:35:57 originally caused by:
16:35:57  GitHub pull request #24046 of commit cfcf2d38eb1a15ed40b217753c414ff902717be3, no merge conflicts.
16:35:57 [EnvInject] - Loading node environment variables.
16:35:57 Building remotely on high_sierra_stadium_2 (high_sierra macos) in workspace /Users/brew/Jenkins/workspace/Homebrew Core Pull Requests/version/high_sierra
16:35:57  > git rev-parse --is-inside-work-tree # timeout=10
16:35:58 Fetching changes from the remote Git repository
16:35:58  > git config remote.origin.url https://github.com/Homebrew/homebrew-core/ # timeout=10
16:35:58 Fetching upstream changes from https://github.com/Homebrew/homebrew-core/
16:35:58  > git --version # timeout=10
16:35:58  > git fetch --tags --progress https://github.com/Homebrew/homebrew-core/ +refs/pull/*:refs/remotes/origin/pr/*
16:36:48 ERROR: Error fetching remote repo 'origin'
16:36:48 hudson.plugins.git.GitException: Failed to fetch from https://github.com/Homebrew/homebrew-core/
16:36:48 	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:862)
16:36:48 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1129)
16:36:48 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1160)
16:36:48 	at hudson.scm.SCM.checkout(SCM.java:495)
16:36:48 	at hudson.model.AbstractProject.checkout(AbstractProject.java:1202)
16:36:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:574)
16:36:48 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
16:36:48 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
16:36:48 	at hudson.model.Run.execute(Run.java:1724)
16:36:48 	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
16:36:48 	at hudson.model.ResourceController.execute(ResourceController.java:97)
16:36:48 	at hudson.model.Executor.run(Executor.java:421)
16:36:48 Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/Homebrew/homebrew-core/ +refs/pull/*:refs/remotes/origin/pr/*" returned status code 128:
16:36:48 stdout: 
16:36:48 stderr: error: RPC failed; curl 18 transfer closed with outstanding read data remaining
16:36:48 fatal: The remote end hung up unexpectedly
16:36:48 
16:36:48 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1990)
16:36:48 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1709)
16:36:48 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:72)
16:36:48 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:400)
16:36:48 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:153)
16:36:48 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:146)
16:36:48 	at hudson.remoting.UserRequest.perform(UserRequest.java:207)
16:36:48 	at hudson.remoting.UserRequest.perform(UserRequest.java:53)
16:36:48 	at hudson.remoting.Request$2.run(Request.java:358)
16:36:48 	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
16:36:48 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
16:36:48 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
16:36:48 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:36:48 	at java.lang.Thread.run(Thread.java:748)
16:36:48 	Suppressed: hudson.remoting.Channel$CallSiteStackTrace: Remote call to high_sierra_stadium_2
16:36:48 		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1693)
16:36:48 		at hudson.remoting.UserResponse.retrieve(UserRequest.java:310)
16:36:48 		at hudson.remoting.Channel.call(Channel.java:908)
16:36:48 		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:146)
16:36:48 		at sun.reflect.GeneratedMethodAccessor416.invoke(Unknown Source)
16:36:48 		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
16:36:48 		at java.lang.reflect.Method.invoke(Method.java:498)
16:36:48 		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:132)
16:36:48 		at com.sun.proxy.$Proxy80.execute(Unknown Source)
16:36:48 		at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:860)
16:36:48 		at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1129)
16:36:48 		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1160)
16:36:48 		at hudson.scm.SCM.checkout(SCM.java:495)
16:36:48 		at hudson.model.AbstractProject.checkout(AbstractProject.java:1202)
16:36:48 		at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:574)
16:36:48 		at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
16:36:48 		at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
16:36:48 		at hudson.model.Run.execute(Run.java:1724)
16:36:48 		at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
16:36:48 		at hudson.model.ResourceController.execute(ResourceController.java:97)
16:36:48 		at hudson.model.Executor.run(Executor.java:421)
16:36:48 ERROR: Error fetching remote repo 'origin'
16:36:48 Retrying after 10 seconds
16:36:58  > git rev-parse --is-inside-work-tree # timeout=10
16:36:58 Fetching changes from the remote Git repository
16:36:58  > git config remote.origin.url https://github.com/Homebrew/homebrew-core/ # timeout=10
16:36:58 Fetching upstream changes from https://github.com/Homebrew/homebrew-core/
16:36:58  > git --version # timeout=10
16:36:58  > git fetch --tags --progress https://github.com/Homebrew/homebrew-core/ +refs/pull/*:refs/remotes/origin/pr/*
16:37:21 Checking out Revision d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124 (refs/remotes/origin/pr/24046/merge)
16:37:21  > git config core.sparsecheckout # timeout=10
16:37:21  > git checkout -f d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124
16:37:21 FATAL: Could not checkout d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124
16:37:21 hudson.plugins.git.GitException: Command "git checkout -f d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124" returned status code 128:
16:37:21 stdout: 
16:37:21 stderr: fatal: reference is not a tree: d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124
16:37:21 
16:37:21 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1990)
16:37:21 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$800(CliGitAPIImpl.java:72)
16:37:21 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$9.execute(CliGitAPIImpl.java:2289)
16:37:21 Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to high_sierra_stadium_2
16:37:21 		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1693)
16:37:21 		at hudson.remoting.UserResponse.retrieve(UserRequest.java:310)
16:37:21 		at hudson.remoting.Channel.call(Channel.java:908)
16:37:21 		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:146)
16:37:21 		at sun.reflect.GeneratedMethodAccessor416.invoke(Unknown Source)
16:37:21 		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
16:37:21 		at java.lang.reflect.Method.invoke(Method.java:498)
16:37:21 		at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:132)
16:37:21 		at com.sun.proxy.$Proxy81.execute(Unknown Source)
16:37:21 		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1210)
16:37:21 		at hudson.scm.SCM.checkout(SCM.java:495)
16:37:21 		at hudson.model.AbstractProject.checkout(AbstractProject.java:1202)
16:37:21 		at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:574)
16:37:21 		at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
16:37:21 		at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
16:37:21 		at hudson.model.Run.execute(Run.java:1724)
16:37:21 		at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
16:37:21 		at hudson.model.ResourceController.execute(ResourceController.java:97)
16:37:21 		at hudson.model.Executor.run(Executor.java:421)
16:37:21 Caused: hudson.plugins.git.GitException: Could not checkout d7b14dd0e3ab34e6e6b1e2b7716b1247f0c17124
16:37:21 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$9.execute(CliGitAPIImpl.java:2313)
16:37:21 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:153)
16:37:21 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:146)
16:37:21 	at hudson.remoting.UserRequest.perform(UserRequest.java:207)
16:37:21 	at hudson.remoting.UserRequest.perform(UserRequest.java:53)
16:37:21 	at hudson.remoting.Request$2.run(Request.java:358)
16:37:21 	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
16:37:21 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
16:37:21 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
16:37:21 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
16:37:21 	at java.lang.Thread.run(Thread.java:748)
16:37:21 Recording test results
16:37:21 ERROR: Step ‘Publish JUnit test result report’ failed: No test report files were found. Configuration error?
16:37:21 Archiving artifacts
16:37:21 [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
16:37:21 Finished: FAILURE
```

Is the homebrew server having problems?

**Comment by ilovezfs** [comment]
No it's an intermittent issue we'll never be able to escape unfortunately.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ylluminarious** [comment]
Oh, ok.

**Comment by ylluminarious** [comment]
@ilovezfs Looks like everything's good, unless you want more changes.

File: Formula/dashing.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Dashing < Formula
+  desc "Generate Dash documentation from HTML files"
+  homepage "https://github.com/technosophos/dashing"
+  url "https://github.com/technosophos/dashing/archive/0.3.0.tar.gz"
+  sha256 "f6569f3df80c964c0482e7adc1450ea44532d8da887091d099ce42a908fc8136"
+  head "https://github.com/technosophos/dashing.git"
+
+  depends_on "go" => :build
+  depends_on "glide" => :build
```
Comment (ilovezfs) [inline Formula/dashing.rb]: alphabetize

File: Formula/dashing.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class Dashing < Formula
+  desc "Generate Dash documentation from HTML files"
+  homepage "https://github.com/technosophos/dashing"
+  url "https://github.com/technosophos/dashing/archive/0.3.0.tar.gz"
+  sha256 "f6569f3df80c964c0482e7adc1450ea44532d8da887091d099ce42a908fc8136"
+  head "https://github.com/technosophos/dashing.git"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: let's not have this

Comment (ylluminarious) [reply Formula/dashing.rb]: Which part are to talking about?

File: Formula/dashing.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    (buildpath/"src/github.com/technosophos/dashing").install buildpath.children
+    cd "src/github.com/technosophos/dashing" do
+      system "glide", "install"
+      system "make", "build"
+      bin.install "dashing"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: prefix.install_metafiles

File: Formula/dashing.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/dashing.rb]: head

Comment (ylluminarious) [reply Formula/dashing.rb]: Ok

File: Formula/dashing.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on "glide" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    (buildpath/"src/github.com/technosophos/dashing").install buildpath.children
+    cd "src/github.com/technosophos/dashing" do
+      system "glide", "install"
+      system "make", "build"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: From the Makefile
```

VERSION := $(shell git describe --tags)

build:
	go build -o dashing -ldflags "-X main.version=${VERSION}" dashing.go
```
That won't work unless we use the tag+revision. But I'd prefer we just run `go build` manually and keep using the tarball

File: Formula/dashing.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      bin.install "dashing"
+    end
+  end
+
+  test do
+    system "#{bin}/dashing", "create"
+    File.exist? "dashing.json"
```
Comment (ilovezfs) [inline Formula/dashing.rb]:
```
assert_predicate testpath/"dashing.json", :exist?
```

**Comment by ilovezfs** [comment]
I'd prefer a test that's a bit more extensive.

**Comment by ylluminarious** [comment]
@ilovezfs In what way? What else should it look for?

**Comment by ilovezfs** [comment]
>Generate Dash documentation from HTML files

So maybe ... generate dash from an html file and verify it's valid?

**Comment by ylluminarious** [comment]
@ilovezfs Hmm, ok, I'll give it a shot.

**Comment by ylluminarious** [comment]
@ilovezfs I did the changes you requested plus a somewhat better test.

File: Formula/dashing.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    (buildpath/"src/github.com/technosophos/dashing").install buildpath.children
+    cd "src/github.com/technosophos/dashing" do
+      system "glide", "install"
+      system "go", "build", "-o", bin/"peco", "dashing.go"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: also need to pass -ldflags "-X main.version=#{version}" dashing.go

File: Formula/dashing.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    # download ruby docs just as dummy documentation to test with
+    system "curl", "-OL", "https://ruby-doc.org/downloads/ruby_2_5_0_core_rdocs.tgz"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: use a resource block to get this

File: Formula/dashing.rb (hunk 6 above)
Comment (ylluminarious) [reply Formula/dashing.rb]: @ilovezfs Ok, I implemented that.

File: Formula/dashing.rb (hunk 7 above)
Comment (ylluminarious) [reply Formula/dashing.rb]: @ilovezfs How are you supposed to do that in a test only?

Comment (ilovezfs) [reply Formula/dashing.rb]: Grep the formulae for `testpath.install` :)

Comment (ylluminarious) [reply Formula/dashing.rb]: @ilovezfs I ended up using `resource.stage`, but thanks for pointing me in the right direction :)

**Comment by ylluminarious** [comment]
@ilovezfs Anything else?

File: Formula/dashing.rb (hunk 6 above)
Comment (ylluminarious) [reply Formula/dashing.rb]: Hmm, I didn't realize, but `brew audit` doesn't like what I did by putting the whole command on one line. Unfortunately, `go build` doesn't like the way the `brew audit` wants me to do it. What am I supposed to do?

File: Formula/dashing.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    (buildpath/"src/github.com/technosophos/dashing").install buildpath.children
+    cd "src/github.com/technosophos/dashing" do
+      system "glide", "install"
+      system "go build -o #{bin}/dashing -ldflags \"-X main.version=#{version}\" dashing.go"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    resource("ruby_docs_tarball").stage("#{testpath}/ruby_2_5_0_core")
```
Comment (ilovezfs) [inline Formula/dashing.rb]: `testpath.install resource("ruby_docs_tarball")` and no need to `cd` below.

File: Formula/dashing.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    (buildpath/"src/github.com/technosophos/dashing").install buildpath.children
+    cd "src/github.com/technosophos/dashing" do
+      system "glide", "install"
+      system "go build -o #{bin}/dashing -ldflags \"-X main.version=#{version}\" dashing.go"
```
Comment (ilovezfs) [inline Formula/dashing.rb]: grep for `-X main.version` in formulae :)

Comment (ylluminarious) [reply Formula/dashing.rb]: Doh! Sorry for my noobishness.

**Comment by ylluminarious** [comment]
Ok, we're finally green...

**Review by ilovezfs — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thank you @ylluminarious for the pull request!

**Comment by ylluminarious** [comment]
@fxcoudert @ilovezfs You're welcome! And thank you as well.

**Pull request closed without merging**